edelphi/src/main/java/fi/metatavu/edelphi/pages/panel/admin/report/thesis/ThesisTimelineQueryReportPage.java

```diff
@@ -93,31 +96,49 @@ public Chart constructChart(ChartContext chartContext, QueryPage queryPage) {
         int y = (int) ((yValues.get(i) - min) / step);
         values[x][y] = new Double(values[x][y] != null ? values[x][y] + 1 : 1);
       }
+    
       return ChartModelProvider.createBubbleChart(queryPage.getTitle(), null, captions, null, captions, 0, 0, values);
-    }
-    else {
+    } else {
       QueryField queryField = queryFieldDAO.findByQueryPageAndName(queryPage, "timeline.value1");
-      List<Double> data = ReportUtils.getNumberFieldData(queryField, queryReplies);
       String chartTitle = QueryPageUtils.getSetting(queryPage, "timeline.value1Label"); 
-      Map<Double, String> dataNames = new HashMap<Double, String>();
-      List<Double> occurences = new ArrayList<Double>();
+      
+      List<Double> data = ReportUtils.getNumberFieldData(queryField, queryReplies);
+      List<Double> occurences = new ArrayList<>();
       Map<Double, Long> classifiedData = ReportUtils.getClassifiedNumberFieldData(data);
       
       for (double d = min; d <= max; d += step) {
-        String caption = step % 1 == 0 ? new Long(Math.round(d)).toString() : new Double(d).toString();
-        dataNames.put(d, caption);
         occurences.add(classifiedData.get(d) == null ? 0d : classifiedData.get(d));
       }
+
+      QueryFieldDataStatistics statistics = createStatistics(data, min, max, step);
+      Double avg = statistics.getCount() >= AVG_MIN_COUNT ? statistics.getAvg() : null;
+      Double q1 = statistics.getCount() >= QUARTILE_MIN_COUNT ? statistics.getQ1() : null;
+      Double q3 = statistics.getCount() >= QUARTILE_MIN_COUNT ? statistics.getQ3() : null;
       
-      
-      QueryFieldDataStatistics statistics = ReportUtils.getStatistics(data, dataNames);
-      Double avg = statistics.getCount() > 1 ? statistics.getAvg() : null;
-      Double q1 = statistics.getCount() >= 5 ? statistics.getQ1() : null;
-      Double q3 = statistics.getCount() >= 5 ? statistics.getQ3() : null;
       return ChartModelProvider.createBarChart(chartTitle, null, captions, occurences, avg, q1, q3);
     }
   }
   
+  /**
+   * Creates statistics object
+   * 
+   * @param data data
+   * @param min min 
+   * @param max max
+   * @param step step
+   * @return statistics object
+   */
+  private QueryFieldDataStatistics createStatistics(List<Double> data, double min, double max, double step) {
+    Map<Double, String> dataNames = new HashMap<>();
+    
+    for (double d = min; d <= max; d += step) {
+      String caption = step % 1 == 0 ? new Long(Math.round(d)).toString() : new Double(d).toString();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "Long.toString" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use "Double.toString" instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "Long" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this "Double" constructor [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2129)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Call the static method Long.toString(...) instead of instantiating a temporary object to perform this to string conversion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1158)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Call the static method Double.toString(...) instead of instantiating a temporary object to perform this to string conversion. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1158)